compiler: remove --lzma option from upx call

nsauk opened the pull request:
UPX can throw NotCompressibleException for some reasons, in this case compiler says that we haven't installed it and willn't use compression at all. Without using --lzma flag I've succesfully compressed the same binary file.
I'm not sure about what to do with tests, at this time I haven't removed --lzma from test code.

Maybe we need public discussion about default preferences for UPX.

spytheman commented:
upx --lzma produces the smallest binaries while only using a little more time than plain upx, and significantly less time than upx --ultra-brute:

```shell
0[09:36:29] /v/nv LINUX $ ./v -o v v.v
0[09:36:35] /v/nv LINUX $ cp v v_upx_brute; cp v v_upx_lzma; cp v v_upx;
0[09:36:47] /v/nv LINUX $ upx -qqq --ultra-brute v_upx_brute; upx -qqq --lzma v_upx_lzma; upx -qqq v_upx
0[09:37:08] /v/nv LINUX $ ls -larS v v_upx v_upx_lzma v_upx_brute
-rwxrwxr-x 1 delian delian 172652 Jan 22 09:36 v_upx_lzma
-rwxrwxr-x 1 delian delian 172652 Jan 22 09:36 v_upx_brute
-rwxrwxr-x 1 delian delian 228500 Jan 22 09:36 v_upx
-rwxrwxr-x 1 delian delian 791336 Jan 22 09:36 v
0[09:37:13] /v/nv LINUX $ upx --version
upx 3.91
UCL data compression library 1.03
LZMA SDK version 9.22 beta
Copyright (C) 1996-2013 Markus Franz Xaver Johannes Oberhumer
Copyright (C) 1996-2013 Laszlo Molnar
Copyright (C) 2000-2013 John F. Reiser
Copyright (C) 2002-2013 Jens Medoch
Copyright (C) 1999-2006 Igor Pavlov
UPX comes with ABSOLUTELY NO WARRANTY; for details type 'upx -L'.
0[09:37:19] /v/nv LINUX $
```

```shell
0[09:37:19] /v/nv LINUX $ cp v v_upx_brute; cp v v_upx_lzma; cp v v_upx;
0[09:38:33] /v/nv LINUX $ time upx -qqq --ultra-brute v_upx_brute

real    0m7.102s
user    0m7.028s
sys     0m0.028s
0[09:38:48] /v/nv LINUX $ time upx -qqq --lzma v_upx_lzma

real    0m0.619s
user    0m0.605s
sys     0m0.010s
0[09:38:51] /v/nv LINUX $ time upx -qqq v_upx

real    0m0.253s
user    0m0.244s
sys     0m0.005s
0[09:38:56] /v/nv LINUX $
```

spytheman commented:
What version of UPX are you using, and on what binary @nsauk ?

spytheman commented:
We can make the compiler first try `upx --lzma -qqq` and if that fails, then try just plain `upx -qqq`.

nsauk commented:
@spytheman I use UPX 3.95. Source code:
```$ cat uptime.v
import os
import term

fn main() {
  uptime := os.exec('uptime')?
  days := uptime.output.split('up ')[1].split(',')[0]
  println('Uptime: ' + term.bold(term.bright_cyan(days)))
}```

Try to compile smallest binary:
```v build -prod -compress uptime.v
upx failed
install upx```

Built on this step file has size 18544 bytes.

Then try compress it with `--lzma`:
```upx --lzma -qq uptime
upx: uptime: NotCompressibleException```

Then try it compress without any opts:
```upx -qq uptime
     18544 ->      8980   48.43%   linux/amd64   uptime```

Now it's just 8980 bytes!

Now let's compare size of binary if we use `v build` without `-prod`, just compress (calls `upx --lzma`):
``` v build -compress uptime.v && du -b uptime
31312	uptime```

Of course, my source code example look like joke, but it shows what may happens when using UPX.

spytheman commented:
That is because LZMA is a dictionary compressor, and it embeds the dictionary in the final compressed executable. For very small programs, that can increase them. 
Try to compress some of the UI examples, or v itself.

For me:
```shell
0[07:06:03] /v/vui/examples LINUX $ ls -lart users*
-rw-rw-r-- 1 delian delian   4183 яну 19 14:45 users.v
-rwxrwxr-x 1 delian delian 101952 яну 22 18:30 users.upx.lzma.395
-rwxrwxr-x 1 delian delian 101512 яну 22 18:30 users.upx.lzma
-rwxrwxr-x 1 delian delian 101952 яну 22 18:30 users.upx.brute.395
-rwxrwxr-x 1 delian delian 101512 яну 22 18:30 users.upx.brute
-rwxrwxr-x 1 delian delian 118192 яну 22 18:30 users.upx.395
-rwxrwxr-x 1 delian delian 126612 яну 22 18:30 users.upx
-rwxrwxr-x 1 delian delian 458264 яну 22 18:30 users.non-stripped
-rwxrwxr-x 1 delian delian 366672 яну 22 18:30 users
0[07:06:09] /v/vui/examples LINUX $
```

spytheman commented:
For your small program, I get:
```shell
0[07:20:31] /v/nv LINUX $ for i in vuptime; do ./v -prod -o $i.non-stripped $i.v; cp $i.non-stripped $i; strip $i; cp $i $i.upx; cp $i $i.upx.lzma ; cp $i $i.upx.brute; cp $i $i.upx.395; cp $i $i.upx.lzma.395; cp $i $i.upx.brute.395;   upx --lzma -qqq *.lzma ;  upx --ultra-brute -qqq *.brute  ;  upx -qqq *.upx;  /opt/upx-3.95/upx  -qqq *.upx.395 ;  /opt/upx-3.95/upx --lzma -qqq  *.lzma.395 ; /opt/upx-3.95/upx --ultra-brute -qqq *.upx.brute.395  ; ls -larS ${i}*; done
-rw-rw-r-- 1 delian delian   180 яну 22 17:55 vuptime.v
-rwxrwxr-x 1 delian delian  8620 яну 23 07:20 vuptime.upx.brute
-rwxrwxr-x 1 delian delian  8648 яну 23 07:20 vuptime.upx
-rwxrwxr-x 1 delian delian  8740 яну 23 07:20 vuptime.upx.brute.395
-rwxrwxr-x 1 delian delian  8748 яну 23 07:20 vuptime.upx.395
-rwxrwxr-x 1 delian delian 10336 яну 23 07:20 vuptime.upx.lzma
-rwxrwxr-x 1 delian delian 10660 яну 23 07:20 vuptime.upx.lzma.395
-rwxrwxr-x 1 delian delian 14568 яну 23 07:20 vuptime
-rwxrwxr-x 1 delian delian 18000 яну 23 07:20 vuptime.non-stripped
0[07:20:32] /v/nv LINUX $
```

spytheman commented:
By the way, why do you do `v build yourfile.v`? 
Is not `v yourfile.v` enough to compile it?

medvednikov commented:
> An interesting thing about v, is that unlike for other example programs, I get even smaller binaries when I compress non production builds:

I guess sometimes production builds can result in more instructions.

For example, I saw clang optimize a `for i < 10 {` loop to 10 repeated operations.
